lib/rspec/core/example.rb

```diff
@@ -565,13 +566,23 @@ def example_skipped?
         # @api private
         # Records the finished status of the example.
         def record_finished(status, finished_at)
-          self.status      = status
-          self.finished_at = finished_at
-          self.run_time    = (finished_at - started_at).to_f
+          self.status = status
+          calculate_run_time(finished_at)
+        end
+
+        # @api private
+        # Populates finished_at and run_time if it has not yet been set
+        def ensure_timing_set(clock)
+          finished_at || calculate_run_time(clock.now)
```

Comment:
This works, but feels like an abuse of `||`.  Typically `||` is used to return one of two values but here we don't care about the return value -- this method is a _command_ method, not a _query_ method, and at the call site the return value is ignored.  Using `||` works because it short-circuits and does not evaluate the right expression if `finished_at` is set...but that feels like an abuse of `||`.  I think it'd be far more clear if written as:

``` ruby
calculate_run_time(clock.now) unless finished_at
```

After all, the goal of this method is to calculate the run time unless `finished_at` is already set -- and using `unless` allows us to state that more directly.